### ISSUE TITLE
nextercism: implement workspace command

### ISSUE THREAD

**Issue by kytrinyx** [exercism/cli#413]
_To be merged into the nextercism branch in https://github.com/exercism/cli/pull/410_

Command name: `workspace` (alias `w`).
File: `cmd/workspace.go`.

This outputs the path to the user's workspace directory.

To get this value, load up a `config.NewUserConfig()`, which has a `Workspace` field.

**Comment by kytrinyx** [comment]
@exercism/windows on Linux/MacOS you can say

    cd `exercism workspace`

or 

    cd $(exercism workspace)

To change directory to the exercism workspace. How would you do this on Windows?

**Comment by NobbZ** [comment]
`cmd.exe` users won't have an alternative available. And as far as my google foo goes powershell should be able to use `$()` as well.

**Comment by kytrinyx** [comment]
Being able to give a Powershell command would be grand, thanks!

**Comment by rpottsoh** [comment]
> To change directory to the exercism workspace. How would you do this on Windows?

An environment variable may be used perhaps?
```
set WorkSpace="c:\my exercism workspace"
```
You may change to that directory like so in cmd:
```
cd %WorkSpace%
```

As far as PS goes I'm no expert.  A quick search did yield [this](https://social.msdn.microsoft.com/Forums/silverlight/en-US/1817d043-6477-4719-b0d8-c4b93a2df1b5/using-powershell-to-create-and-map-workspaces?forum=tfsversioncontrol) which I hope is helpful.

Is this sort of what you are asking?

**Comment by kytrinyx** [comment]
> Is this sort of what you are asking?

Yepp, something like that. The basic idea is to give people an easy way to get to their workspace, without having to type (or remember) long paths.

    cd `exercism w`

An env variable is another way of doing this (as are aliases). I'm hoping to find a really nice and easy way to use the `exercism w` to let Windows users do the same thing.

**Comment by rpottsoh** [comment]
So if I am understanding the intent of `exercism w` correctly it might look something like this:
```
C:\Program Files>
C:\Program Files> exercism w
C:\My Exercism Workspace>
```
`Program Files` might be the landing folder when cmd is executed.

**Comment by kytrinyx** [comment]
@rpottsoh I don't think that is possible, since the command `exercism w` would be executed in a subshell. We'd need something to take the output of that command, and then change directory to it.

If this isn't a common idiom on Windows, then it might not be worth contorting ourselves to make it so :)

**Comment by NobbZ** [comment]
Its not a common idiom among windows users that still do not use powershell but stick to `cmd.exe` ;)

“Composition” of commands seems to be done in powershell quite often, though before googling around I've only seen piping.

For windows we will have at least one problem though. The working dir is per drive and shell session. So when we are on drive `D:` and the exercism workspace is on drive `C:` and we do `cd $(exercism w)`, it will appear as if nothing happened, until one switches the drive manually… This behaviour fooled me quite often…

**Comment by robphoenix** [comment]
On Powershell you can do `cd $(exercism w)`, I couldn't get anything to work on `cmd.exe`.

**Comment by kytrinyx** [comment]
Thanks for your help figuring out the Windows thing, everyone! I've documented with some caveats in the usage for the command.

Closed by #448
